File: monitor_display.py
#!/usr/bin/env python3
"""
Windows Multi-Monitor Display App
各モニタにそれぞれのID番号と名前を大きく表示
"""

import tkinter as tk
from screeninfo import get_monitors
import sys
import threading

def get_monitor_name_from_windows(index):
    """Windows APIからモニタ名を取得"""
    try:
        import winreg
        reg = winreg.ConnectRegistry(None, winreg.HKEY_LOCAL_MACHINE)
        key = winreg.OpenKey(reg, r'SYSTEM\CurrentControlSet\Enum\DISPLAY')

        for i in range(winreg.QueryInfoKey(key)[0]):
            subkey_name = winreg.EnumKey(key, i)
            subkey = winreg.OpenKey(key, subkey_name)
            try:
                value = winreg.QueryValueEx(subkey, 'FriendlyName')
                if value:
                    return value[0]
            except:
                pass
            winreg.CloseKey(subkey)
        winreg.CloseKey(key)
    except:
        pass
    return f"Monitor {index + 1}"

def create_monitor_window(monitor_index, monitor):
    """モニタごとにウィンドウを作成・表示"""
    root = tk.Tk()
    root.attributes('-fullscreen', True)
    root.configure(bg='#1a1a2e')

    monitor_name = get_monitor_name_from_windows(monitor_index)

    frame = tk.Frame(root, bg='#1a1a2e')
    frame.pack(expand=True, fill=tk.BOTH)

    label_id = tk.Label(
        frame,
        text=f"Monitor {monitor_index + 1}",
        font=('Arial', 120, 'bold'),
        fg='#00d4ff',
        bg='#1a1a2e'
    )
    label_id.pack(pady=50)

    label_name = tk.Label(
        frame,
        text=monitor_name,
        font=('Arial', 60),
        fg='#00ff88',
        bg='#1a1a2e'
    )
    label_name.pack(pady=30)

    label_resolution = tk.Label(
        frame,
        text=f"{monitor.width} × {monitor.height}",
        font=('Arial', 40),
        fg='#ffaa00',
        bg='#1a1a2e'
    )
    label_resolution.pack(pady=20)

    # ESCキーで閉じる
    root.bind('<Escape>', lambda e: root.destroy())

    # ウィンドウをそのモニタに配置
    root.geometry(f"+{monitor.x}+{monitor.y}")

    root.mainloop()

def main():
    monitors = get_monitors()

    if not monitors:
        print("モニタが検出されません")
        sys.exit(1)

    print(f"{len(monitors)}個のモニタを検出しました")

    threads = []
    for i, monitor in enumerate(monitors):
        print(f"Monitor {i}: {monitor}")
        thread = threading.Thread(target=create_monitor_window, args=(i, monitor))
        thread.daemon = False
        thread.start()
        threads.append(thread)

    for thread in threads:
        thread.join()

if __name__ == '__main__':
    main()

File: MonitorDisplay.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Windows.Forms;

namespace WindowsMonitorDisplay
{
    static class Program
    {
        [STAThread]
        static void Main()
        {
            Application.EnableVisualStyles();

            var screens = Screen.AllScreens;

            if (screens.Length == 0)
            {
                MessageBox.Show("No monitors detected.", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return;
            }

            Console.WriteLine(string.Format("Detected {0} monitor(s)", screens.Length));

            List<System.Threading.Thread> threads = new List<System.Threading.Thread>();

            for (int i = 0; i < screens.Length; i++)
            {
                int index = i;
                Screen screen = screens[i];

                System.Threading.Thread thread = new System.Threading.Thread(() =>
                {
                    MonitorForm form = new MonitorForm(index, screen);
                    Application.Run(form);
                });

                thread.Start();
                threads.Add(thread);
            }

            foreach (var thread in threads)
            {
                thread.Join();
            }
        }

        static Dictionary<int, string> GetMonitorFriendlyNames()
        {
            var result = new Dictionary<int, string>();

            try
            {
                var regKey = Microsoft.Win32.Registry.LocalMachine.OpenSubKey(
                    @"SYSTEM\CurrentControlSet\Enum\DISPLAY");

                if (regKey != null)
                {
                    int index = 0;
                    foreach (var subKeyName in regKey.GetSubKeyNames())
                    {
                        var subKey = regKey.OpenSubKey(subKeyName);
                        if (subKey != null)
                        {
                            var friendlyName = subKey.GetValue("FriendlyName");
                            if (friendlyName != null)
                            {
                                result[index] = friendlyName.ToString();
                            }
                            else
                            {
                                result[index] = string.Format("Monitor {0}", index + 1);
                            }
                            index++;
                        }
                    }
                }
            }
            catch { }

            return result;
        }

        public static string GetMonitorName(int index)
        {
            var names = GetMonitorFriendlyNames();
            if (names.ContainsKey(index))
                return names[index];
            return string.Format("Monitor {0}", index + 1);
        }
    }

    public class MonitorForm : Form
    {
        private int _monitorIndex;
        private Screen _screen;

        public MonitorForm(int index, Screen screen)
        {
            _monitorIndex = index;
            _screen = screen;

            this.FormBorderStyle = FormBorderStyle.None;
            this.WindowState = FormWindowState.Maximized;
            this.BackColor = Color.FromArgb(26, 26, 46); // #1a1a2e
            this.TopMost = false;

            // Move to specific monitor
            this.StartPosition = FormStartPosition.Manual;
            this.Location = screen.Bounds.Location;
            this.Size = screen.Bounds.Size;

            // Create controls
            Label labelIndex = new Label
            {
                Text = string.Format("Monitor {0}", _monitorIndex + 1),
                Font = new Font("Arial", 120, FontStyle.Bold),
                ForeColor = Color.FromArgb(0, 212, 255), // #00d4ff
                BackColor = Color.FromArgb(26, 26, 46),
                TextAlign = ContentAlignment.MiddleCenter,
                Dock = DockStyle.None,
                AutoSize = false,
                Width = screen.Bounds.Width,
                Height = 200,
                Top = 100
            };

            string monitorName = Program.GetMonitorName(_monitorIndex);
            Label labelName = new Label
            {
                Text = monitorName,
                Font = new Font("Arial", 60),
                ForeColor = Color.FromArgb(0, 255, 136), // #00ff88
                BackColor = Color.FromArgb(26, 26, 46),
                TextAlign = ContentAlignment.MiddleCenter,
                Dock = DockStyle.None,
                AutoSize = false,
                Width = screen.Bounds.Width,
                Height = 150,
                Top = 320
            };

            string resolution = string.Format("{0} × {1}", screen.Bounds.Width, screen.Bounds.Height);
            Label labelResolution = new Label
            {
                Text = resolution,
                Font = new Font("Arial", 40),
                ForeColor = Color.FromArgb(255, 170, 0), // #ffaa00
                BackColor = Color.FromArgb(26, 26, 46),
                TextAlign = ContentAlignment.MiddleCenter,
                Dock = DockStyle.None,
                AutoSize = false,
                Width = screen.Bounds.Width,
                Height = 100,
                Top = 500
            };

            this.Controls.Add(labelIndex);
            this.Controls.Add(labelName);
            this.Controls.Add(labelResolution);

            this.KeyPreview = true;
            this.KeyDown += (s, e) =>
            {
                if (e.KeyCode == Keys.Escape)
                {
                    this.Close();
                    e.Handled = true;
                }
            };

            Console.WriteLine(string.Format("Monitor {0}: {1} - {2}", _monitorIndex, screen.Bounds, monitorName));
        }
    }
}
